### ISSUE TITLE
Colon rule is triggered when present in a regex literal

### ISSUE THREAD

**Issue by jjgp** [realm/SwiftLint#4358]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Given a regex literal with a colon in it such as:
```Swift
let httpRegex = /(http|https):\/\//
```

I expect that the following warning in the output would not be issued:
```
/Users/jasonprasad/Workspace/PracticeSwiftUI/iOS/Reddit/RedditModels.swift:64:38: warning: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
```

### Environment

* SwiftLint version: 0.49.1
* Installation method used: Homebrew
* Xcode version: 14.0.1
* Paste your configuration file:

```yml
nesting:
  type_level:
    warning: 2
    error: 3
```

**Comment by jpsim** [comment]
Please try with `main` as there were several changes since 0.49.1 that could have fixed this.
